webui/react/src/components/kit/LineChart.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useMemo } from 'react';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { FixedSizeGrid } from 'react-window';
+import { AlignedData } from 'uplot';
+
+import { SyncProvider } from 'components/UPlot/SyncableBounds';
+import UPlotChart, { Options } from 'components/UPlot/UPlotChart';
+import { tooltipsPlugin } from 'components/UPlot/UPlotChart/tooltipsPlugin';
+import { glasbeyColor } from 'shared/utils/color';
+import { Scale } from 'types';
+
+import css from './LineChart.module.scss';
+
+interface Serie {
+  color?: string;
+  data: number[][];
+  name?: string;
+}
+
+interface Props {
+  focusedSeries?: number;
+  height?: number;
+  scale?: Scale;
+  series: Serie[];
+  showLegend?: boolean;
+  title?: string;
+  xLabel?: string;
+  yLabel?: string;
+}
+
+export const LineChart: React.FC<Props> = ({
+  focusedSeries,
+  height = 400,
+  scale = Scale.Linear,
+  series,
+  showLegend = false,
+  title,
+  xLabel,
+  yLabel,
+}: Props) => {
+  const chartData: AlignedData = useMemo(() => {
+    const xValues: number[] = [];
+    const yValues: Record<string, Record<string, number | null>> = {};
+
+    series.forEach((serie, serieIndex) => {
+      yValues[serieIndex] = {};
+      serie.data.forEach((pt) => {
+        xValues.push(pt[0]);
+        yValues[serieIndex][pt[0]] = Number.isFinite(pt[1]) ? pt[1] : null;
+      });
+    });
+
+    xValues.sort((a, b) => a - b);
+    const yValuesArray: (number | null)[][] = Object.values(yValues).map((yValue) => {
+      return xValues.map((xValue) => (yValue[xValue] != null ? yValue[xValue] : null));
+    });
+
+    return [xValues, ...yValuesArray];
+  }, [series]);
+
+  const chartOptions: Options = useMemo(() => {
+    const plugins = [tooltipsPlugin({ isShownEmptyVal: false })];
```

Comment:
@alexjdetermined was wanting the tooltip to show all the series, not just the hovered one(s).

Further enhancement: highlighting the hovered series in the tooltip.

Reply:
if you have a series A with points at {1,3,5} and series B with points at {0,1,2,3,4,5}
`isShownEmptyVal` is filtering out `null`s, such as value of series A at x=2

the new tooltip (I might want to make it a separate file while we are switching over) does sort the series in it from highest y-value to lowest, and bolds the closest ones (along the y-axis)

<img width="360" alt="Screen Shot 2023-01-18 at 12 55 39 PM" src="https://user-images.githubusercontent.com/643918/213281381-3759d9a0-4cc2-4c24-9390-24daf682d61e.png">